### PR TITLE
Monorepo setup

### PR DESCRIPTION
**Pull request by PaulLeCam**
I haven't updated the tests yet, I'm thinking we could split the `packages/connect` setup more by extracting the `iframe` and `example` folders and associated dependencies to the `apps` folder. This way the iframe and management apps would have a similar setup as far as the 3id-connect package is concerned.
The main upside I think would be for the tests setup: the `packages/*` would have unit tests and the root folder could have the integration tests including the connect package, iframe and management apps.

@zachferland what do you think please? Hopefully we can address a bunch of the TODOs we left as well in the process?
Obviously this would take some extra time to refactor but I really think it's worth it to make the setup work without hacks/shortcuts so it's easier to make changes in the future.

### CODE REVIEW

**Comment by PaulLeCam** [comment]
@zachferland I finally got everything working and running in CI so it should be OK to start reviewing, but I still need to update the readmes and maybe remove unnecessary dependencies.

**Review by zachferland — APPROVED** [review]
Looks good! a bit hard to follow entire diff, is there any open questions left or anything to look at in more detail? otherwise seems all good

**Comment by PaulLeCam** [comment]
Yeah sorry for the large diff, I think the main thing to check is if the CI artifacts are properly built?
Ideally we should also update some internal logic and add tests for the individual packages, but I think it's better to scope this after merging this branch so it doesn't become a long-running one?
